dot/core/service.go

```diff
@@ -153,12 +155,17 @@ func NewService(cfg *Config) (*Service, error) {
 			babeKill:         babeKill,
 			isBabeAuthority:  true,
 			lock:             chanLock,
-			closed:           false,
 			syncLock:         syncerLock,
 			blockNumOut:      cfg.SyncChan,
 			respOut:          respChan,
 		}
 
+		// thread safe way to change closed status
+		canLock := atomic.CompareAndSwapUint32(&srv.started, 0, 1)
+		if !canLock {
+			return nil, errors.New("[core] Error when trying to change Service status from stopped to started")
```

Comment:
change error to `failed to change started to 1` or something along those lines, to align with error standards

Reply:
wait, yes is true we change the status to 1, but in fact the representation of it is that we change it to started/ stopped. I would argue that the message is more meaningful than saying 0 or 1. Don't you agree ?

Reply:
okay, the point was to align with error standards, so remove `[core]` and change `error when trying` to `failed`

Reply:
roger

Reply:
done